nixos/modules/config/networking.nix

```diff
@@ -11,6 +11,28 @@ let
 
   localhostMultiple = any (elem "localhost") (attrValues (removeAttrs cfg.hosts [ "127.0.0.1" "::1" ]));
 
+  localServicesFile =
+    pkgs.runCommand "etc-services" { preferLocalBuild = true; } (let
+      overrides = lib.pipe cfg.services [
+        (lib.attrsets.mapAttrsToList (name:
+          { port, protocols, aliases }:
+          (map (proto: "${name} ${toString port}/${proto} ${toString aliases}")
+            protocols)))
+        lib.lists.flatten
+        lib.strings.concatLines
+      ];
+      deletions =
+        map (name: "-e '/^${name} .*/d'") (builtins.attrNames cfg.services);
```

Comment:
I am not sure how well we this could be done but it would be nice if we would give some kind of feedback when things don't match.

Reply:
This deletes entries in the default file if they would clash with the name in the new services. I think `/etc/services` is abandonware so we don't have to worry about dropping stuff from it.